pkg/retainer/retainer.go

```diff
@@ -0,0 +1,19 @@
+package retainer
+
+import (
+	"github.com/iotaledger/hive.go/runtime/module"
+	"github.com/iotaledger/iota-core/pkg/model"
+	iotago "github.com/iotaledger/iota.go/v4"
+	"github.com/iotaledger/iota.go/v4/nodeclient/apimodels"
+)
+
+// Retainer keeps and resolves all the information needed in the API and INX.
+type Retainer interface {
```

Comment:
This should not be an engine module imho.
It should be an optional node component since it is not needed for normal node operation.

Reply:
what about core-api responses? like for block metadata, without it we are not able to provide correct statuses. 
Also placing it inside engine seemed correct for chain switching, as data in retainer only makes sense on one chain.

Reply:
Then this is a main component of the protocol and nothing "optional" and we should treat is as such

Reply:
So we can call it resolved, because its not optional.